src/components/AppleMusic/AppleMusic.ts

```diff
@@ -0,0 +1,145 @@
+    axios
+      .get("http://localhost:8889/login")
+      .then((response) => {
+        this.appleMusicApi = appleMusicAxiosInstance(response.data.token);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  searchName = (
+    searchParams: string,
+    options: {
+      filter: ("albums" | "artists" | "playlists" | "tracks")[];
+      limit: number;
+    } = { filter: ["artists", "playlists"], limit: 20 }
+  ): Promise<SearchResponseName> => {
+    return this.appleMusicApi.get(
+      "catalog/" +
+        //ToDo: should be set by the location the Website is opened from
+        "DE" +
+        "/search?term=" +
+        searchParams.replace(" ", "+") +
+        "&limit=" +
+        options.limit +
+        "&types=" +
+        options.filter.toString()
+    );
+  };
+
+  getSearchName = (
+    searchParams: string,
+    callback: (data: any) => void,
+    options: {
+      filter: ("albums" | "artists" | "playlists" | "tracks")[];
+      limit: number;
+    } = { filter: ["artists", "playlists"], limit: 20 }
+  ) => {
+    this.searchName(searchParams, options).then((res) => {
+      callback(this.dataSearchName(res));
+    });
+  };
+
+  dataSearchName = (data: SearchResponseName): SearchNameResponds => {
+    return {
+      service: "applemusic",
+      playlists: data.data.results.playlists.data.map(
+        (value: DataObject<PlaylistAttributes>) => {
+          return {
+            image: value.attributes.artwork?.url
+              .replace("{h}", "220")
+              .replace("{w}", "220"),
+            name: value.attributes.name,
+            nameCreator: value.attributes.curatorName,
+            url: value.attributes.playParams.id,
+          };
+        }
+      ),
+    };
+  };
+
+  playlistById = (playlistId: string): Promise<SearchResponsePlaylist> => {
+    return this.appleMusicApi.get(
+      "catalog/" +
+        //ToDo: should be set by the location the Website is opened from
+        "DE" +
+        "/playlists/" +
+        playlistId
+    );
+  };
+
+  dataPlaylistById = (playlistData: SearchResponsePlaylist): PlaylistObject => {
+    return {
+      image: playlistData.data.data[0].attributes.artwork?.url
+        .replace("{h}", "220")
+        .replace("{w}", "220"),
+      name: playlistData.data.data[0].attributes.name,
+      nameCreator: playlistData.data.data[0].attributes.curatorName,
+      songs: playlistData.data.data[0].relationships.tracks.data.map(
+        (value): Song => {
+          return {
+            name: value.attributes.name,
+            artists: value.attributes.artistName,
+            duration: value.attributes.durationInMillis,
+          };
+        }
+      ),
+      description: playlistData.data.data[0].attributes.description?.short,
+    };
+  };
+
+  getPlaylistById = (playlistId: string, callback: (data: any) => void) => {
+    this.playlistById(playlistId).then((value) => {
+      callback(this.dataPlaylistById(value));
+    });
+  };
```

Comment:
This can be simplified, by combining it into on function.

```suggestion
  dataPlaylistById = ( playlistId: string,  callback:(data: any) => PlaylistObject):
    => {
    this.playlistById(playlistId).then((playlistData:SearchResponsePlaylist) => {callback({
      image: playlistData.data.data[0].attributes.artwork?.url
        .replace("{h}", "220")
        .replace("{w}", "220"),
      name: playlistData.data.data[0].attributes.name,
      nameCreator: playlistData.data.data[0].attributes.curatorName,
      songs: playlistData.data.data[0].relationships.tracks.data.map(
        (value): Song => {
          return {
            name: value.attributes.name,
            artists: value.attributes.artistName,
            duration: value.attributes.durationInMillis,
          };
        }
      ),
      description: playlistData.data.data[0].attributes.description?.short,
    })}
  };
```